CONTRIBUTING.md

```diff
@@ -107,10 +107,85 @@ approved, other developers with commit access may commit it for the author. When
 doing so, it is important to retain correct attribution of the contribution.
 Generally speaking, Git handles attribution automatically.
 
+## Development
+
+### Local Setup
+
+This project uses [Poetry](https://python-poetry.org/) to manage its dependencies. If you are not familiar with this tool, we highly recommend checking out [poetry docs](https://python-poetry.org/docs/) to get used to the basic usage.
+
+Step 1: Install Poetry [here](https://python-poetry.org/docs/#installation) 
+
+Step 2: Run poetry install
+
+Step 3: Optionally Run poetry shell
+
+Additional info if you run into trouble: [Poetry Environments](https://python-poetry.org/docs/managing-environments/)
+
+You could use pyenv with poetry together by following [this section](#setup-with-pyenv)
+
+### Setup with pyenv
+
+It is recommended to use pyenv to switch between Python versions:
+
+Install pyenv [here]https://github.com/pyenv/pyenv#installation)
+
+Install and set python version with pyenv:
+
+```console
+pyenv install 3.7.8
+pyenv local 3.7.8
+```
+
+Note: You are welcome to use any python version >= 3.7.1
```

Comment:
do we want to be more specific on the versions we support?  3.7 to 3.10?

Reply:
sure, just updated, it needs to actually >= 3.7.1 for pandas to work for 3.10